src/hotspot/share/runtime/thread.cpp

```diff
@@ -1859,12 +1859,6 @@ void JavaThread::exit(bool destroy_vm, ExitType exit_type) {
   Handle threadObj(this, this->threadObj());
   assert(threadObj.not_null(), "Java thread object should be created");
 
-  // FIXIT: This code should be moved into else part, when reliable 1.2/1.3 check is in place
-  {
-    EXCEPTION_MARK;
-
-    CLEAR_PENDING_EXCEPTION;
-  }
```

Comment:
I think I agree that this is redundant. Seems to be a leftover from something else.

Reply:
Yes.  Look above it.  It clears the exception mark so there's never an exception pending.  strange we didn't clean this out sometime between 1.2 and now.